File: 1915-number-of-wonderful-substrings/1915-number-of-wonderful-substrings.js
/**
 * @param {string} word
 * @return {number}
 */
var wonderfulSubstrings = function(word) {
    let ans = 0, _xor = 0;
    mp = new Map();
    mp.set(0, 1);

    for(let c of word) {
        let bit = c.charCodeAt(0) - 'a'.charCodeAt(0);
        _xor ^= (1 << bit);
        for(let i = 0; i<10; i++) ans += mp.get(_xor ^ (1 << i)) || 0;
        ans += mp.get(_xor) || 0;
        mp.set(_xor, (mp.get(_xor) || 0) + 1);
    }

    return ans;
};

File: 1915-number-of-wonderful-substrings/1915-number-of-wonderful-substrings.py
class Solution(object):
    def wonderfulSubstrings(self, word):
        ans, _xor = 0, 0
        mp = {}
        mp[0] = 1

        for c in word:
            bit = ord(c) - ord('a')
            _xor ^= (1 << bit)
            for i in range(10):
                ans += mp[_xor ^ (1 << i)] if _xor ^ (1 << i) in mp else 0
            ans += mp[_xor] if _xor in mp else 0
            if _xor in mp:
                mp[_xor] += 1
            else:
                mp[_xor] = 1
        
        return ans